src/nodes.coffee

```diff
@@ -936,15 +936,30 @@ exports.NumberLiteral = class NumberLiteral extends Literal
         @parsedValue = @value
         @value = "#{@value}"
       else
-        @parsedValue = Number @value
+        @parsedValue = parseNumber @value
 
-  astType: -> 'NumericLiteral'
+  isBigInt: ->
+    /n$/.test @value
+
+  astType: ->
+    if @isBigInt()
+      'BigIntLiteral'
```

Comment:
I hadn't realized this before but I guess BigInt's get their own `BigIntLiteral` type in Babel AST's

Reply:
Should we have a separate grammar for them? Rather than overloading NumericLiteral.

Reply:
> Should we have a separate grammar for them? Rather than overloading NumericLiteral

That seems reasonable - I guess that would "trickle down" to needing different token types for BigInt's as well (in order to be able to distinguish them at the grammar level)?

Another option if they only behave differently during AST generation would be to start by only distinguishing them as a new eg `BigIntLiteral` node class during AST generation (rather than at the grammar level) like eg how we're currently only instantiating the `ObjectProperty` node class during AST generation

Reply:
Since a BigInt apparently isn't a Number, I feel like we'll run into trouble long term by conflating them.